### PR TITLE
Added more methods on Ocean API and fixed consume flow.

### PR DESCRIPTION
**Pull request by Pedro-vk**
## Description

Added more methods on Ocean API and fixed consume flow.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
 
## Checklist:

- [x] Follows the code style of this project.
- [x] Tests Cover Changes
- [x] Documentation

#### Funny gif

Put a link of a funny gif inside the parenthesis-->

### CODE REVIEW

File: integration/ocean/ConsumeAsset.test.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        })
+
+        // Accounts
+        publisher = new Account("0x00Bd138aBD70e2F00903268F3Db08f2D25677C9e")
+        publisher.setPassword("node0")
+        consumer = new Account("0x068Ed00cF0441e4829D9784fCBe7b9e26D4BD8d0")
+        consumer.setPassword("secret")
+
+        // Data
+        metadata = {
```
[the remark this answers is not in the document]
Comment (Pedro-vk) [reply integration/ocean/ConsumeAsset.test.ts]: I've it in mind, thanks!

File: src/ocean/Ocean.ts [hunk 2]
```diff
@@ -167,49 +190,7 @@ export default class Ocean {
...
-            .getBrizo()
-            .initializeServiceAgreement(
-                did,
-                serviceAgreementId,
-                serviceDefinitionId,
-                serviceAgreementSignature,
-                consumer.getId())
-
-        await filesPromise
+        return await this.assets.consume(serviceDefinitionId, did, serviceDefinitionId, consumer)
```
Comment (ssallam) [inline src/ocean/Ocean.ts]: First arg should be `agreementId`, no?

File: src/ocean/OceanAssets.ts [hunk 3]
```diff
@@ -181,8 +181,62 @@ export default class OceanAssets {
...
+            throw new Error("Consume asset failed, service definition is missing the `serviceEndpoint`.")
+        }
+
+        const secretStoreUrl = authorizationService.service === "SecretStore" && authorizationService.serviceEndpoint
+        const secretStoreConfig = {
+            secretStoreUri: secretStoreUrl,
+        }
+
+        Logger.log("Decrypting files")
+        const decryptedFiles = await SecretStoreProvider.getSecretStore(secretStoreConfig).decryptDocument(DID.parse(did).getId(), files)
```
Comment (ssallam) [inline src/ocean/OceanAssets.ts]: Need to make sure that SS decyption is using the consumer account.

File: src/ocean/Ocean.ts (hunk 2 above)
Comment (Pedro-vk) [reply src/ocean/Ocean.ts]: I'm going to remove these part, and just throw an error to ensure that is not going to be used. thanks!

File: src/ocean/OceanAssets.ts (hunk 3 above)
Comment (Pedro-vk) [reply src/ocean/OceanAssets.ts]: The problem is that we don't store the password there. How we should handle it?

Comment (Pedro-vk) [reply src/ocean/OceanAssets.ts]: In any case, I'm going to close the PR and continue working on that before

**Pull request merged**